lib/SIL/Utils/BasicBlockUtils.cpp

```diff
@@ -521,3 +521,36 @@ void swift::findJointPostDominatingSet(
     }
   }
 }
+
+//===----------------------------------------------------------------------===//
+//                          checkReachingBlockDominance
+//===----------------------------------------------------------------------===//
+
+#ifndef NDEBUG
+/// Verify that \p sourceBlock dominates \p destBlock given that some path from
+/// \p sourceBlock to \p destBlock exists.
+///
+/// Useful for assertions without computing Dominance.
+bool swift::checkReachingBlockDominates(SILBasicBlock *sourceBlock,
+                                        SILBasicBlock *destBlock) {
+  SILBasicBlock *entryBlock = sourceBlock->getParent()->getEntryBlock();
+  BasicBlockSet visitedBlocks(sourceBlock->getParent());
+  SmallVector<SILBasicBlock *, 32> workList = {destBlock};
```

Comment:
Is this a place where you'd want to use DAGNodeWorklist?

Reply:
`BasicBlockWorklist` should be used here

Reply:
@nate-chandler Yes, DAGNodeWorklist (which should probably be called GraphNodeWorklist), would work. I think the algorithm was originally slightly different.

BasicBlockWorklist also works, so I used that. But you need to be careful about the stack discipline of that data structure and we can't have more than 32 active!